eland/operations.py

```diff
@@ -1233,58 +1233,55 @@ def _es_results(
         if script_fields is not None:
             body["script_fields"] = script_fields
 
-        # Only return requested field_names
+        # Only return requested field_names and add them to body
         _source = query_compiler.get_field_names(include_scripted_fields=False)
-        if _source:
-            # For query_compiler._client.search we could add _source
-            # as a parameter, or add this value in body.
-            #
-            # If _source is a parameter it is encoded into to the url.
-            #
-            # If _source is a large number of fields (1000+) then this can result in an
-            # extremely long url and a `too_long_frame_exception`. Therefore, add
-            # _source to the body rather than as a _source parameter
-            body["_source"] = _source
-        else:
-            body["_source"] = False
-
-        es_results: Any = None
-
-        # If size=None use scan not search - then post sort results when in df
-        # If size>10000 use scan
-        is_scan = False
-        if size is not None and size <= DEFAULT_ES_MAX_RESULT_WINDOW:
-            if size > 0:
-                es_results = query_compiler._client.search(
-                    index=query_compiler._index_pattern,
-                    size=size,
-                    sort=sort_params,
-                    body=body,
-                )
-        else:
-            is_scan = True
-            es_results = scan(
-                client=query_compiler._client,
-                index=query_compiler._index_pattern,
-                query=body,
-            )
-            # create post sort
-            if sort_params is not None:
-                post_processing.append(SortFieldAction(sort_params))
-
-        if is_scan:
-            while True:
-                partial_result, df = query_compiler._es_results_to_pandas(
-                    es_results, collector.batch_size(), collector.show_progress
-                )
-                df = self._apply_df_post_processing(df, post_processing)
-                collector.collect(df)
-                if not partial_result:
-                    break
+        body["_source"] = _source if _source else False
+
+        is_pagination_required: bool = False
+
+        if result_size and (0 <= result_size <= DEFAULT_ES_MAX_RESULT_WINDOW):
+            body["size"] = result_size
         else:
-            partial_result, df = query_compiler._es_results_to_pandas(es_results)
-            df = self._apply_df_post_processing(df, post_processing)
-            collector.collect(df)
+            body["size"] = DEFAULT_ES_MAX_RESULT_WINDOW
+            is_pagination_required = True
+
+        if sort_params is not None:
+            body["sort"] = [sort_params]
+        elif is_pagination_required:
```

Comment:
Since we're always paginating with PIT we should unconditionally add `body[sort]`, no need to do `is_pagination_required` conditional branches.